src/main/java/sparksoniq/spark/iterator/flowr/ReturnClauseSparkIterator.java

```diff
@@ -49,8 +61,66 @@ public ReturnClauseSparkIterator(RuntimeTupleIterator child, RuntimeIterator exp
         return itemRDD;
     }
 
-    private JavaRDD<Item> itemRDD;
-    
-    private RuntimeTupleIterator _child;
+    @Override
+    protected boolean hasNextLocal() {
+        return _hasNext;
+    }
+
+    @Override
+    protected Item nextLocal() {
+        if(_hasNext == true){
+            Item result = _nextLocalResult;  // save the result to be returned
+            setNextLocalResult();            // calculate and store the next result
+            return result;
+        }
+        throw new IteratorFlowException("Invalid next() call in Object Lookup", getMetadata());
+    }
+
+    @Override
+    protected void openLocal(DynamicContext context) {
+        this._currentDynamicContext = context;
+
+        _child.open(context);
+        setNextLocalResult();
+    }
+
+    private void setNextLocalResult() {
+        _nextLocalResult = null;
+        if (_expression.hasNext()) {
+            _nextLocalResult = _expression.next();
+        } else {
+            _expression.close();
+
+            if (_child.hasNext()) {
+                FlworTuple tuple = _child.next();
+                DynamicContext dynamicContext = new DynamicContext(tuple);
```

Comment:
You need to chain it behind _currentDynamicContext (I think there is another constructor for that somewhere) so that all bindings from the outer context are correctly inherited.

Reply:
I have found earlier that there is a mechanism to connect dynamic contexts to each other through a parent field relationship. However in the current implementation this is done through a constructor (no setter). 

Additionally, it is also possible to create a dynamic context from a tuple which is also done through a constructor which calls a private method for the operation.

If we want to connect dynamic contexts with the parent field we either need a setter or expose the tuple -> dynamic context method to outside. I can implement your selection, in this or in a different PR based on your feedback.

Reply:
I think we simply need to add a constructor to the DynamicContext class:

    public DynamicContext(DynamicContext parent, FlworTuple tuple){
            this._parent = parent;
            this._variableValues = new HashMap<>();
            generateDynamicContextFromTupleRdd(tuple);
        }

So far there was only one version to connect to the parent, and one to convert a tuple, but none that did both at the same time.

Reply:
For the record: it is also a viable option, as you suggest, to expose the conversion method to the outside world (maybe calling it setBindingsFromTuple()), because this would avoid constructing a new DynamicContext object for every tuple (performance overhead as object construction is quite expensive): you can created a single DynamicContext object, chained with its parent, and set it in place by calling setBindingsFromTuple for each tuple in turn.

So I am fine either way and leave the decision to you. At some point we will have to profile the code and improve performance anyway.